src/common/lib/util/utils.ts

```diff
@@ -227,14 +227,6 @@ export function forInOwnNonNullProperties(ob: Record<string, unknown>, fn: (prop
   }
 }
 
-/* Useful when the function may mutate the array */
```

Comment:
Is this removal of `safeArrForEach` related to removing code that's supporting older platforms? If not, then I wonder whether it'd be better to keep it; the removed comment makes the intention a bit clearer. Or alternatively this commit could put a comment in `connectionmanager.ts` explaining _why_ we're using `slice()`?

Reply:
It just didn't feel like a good thing to have a separate Utils method just to call two native Array methods back to back without any additional logic.

I do agree though that it is now less clear why we need to use `.slice()` in `connectionmanager.ts`, and adding comments there would mean duplicating that comment, whereas before it was only in one place in Utils.safeArrForEach.

I don't have strong opinion on this one, can omit this commit if you'd prefer that, let me know

Reply:
Decided to add comments to `connectionmanager.ts` in https://github.com/ably/ably-js/pull/1629/commits/8a8e93f3caf4416575d49327cbd549fdf5508e02